### ISSUE TITLE
Investigate how to force dotnet formatting strictness

### ISSUE THREAD

**Issue by hburn7** [osu-tournament-rating/otr-api#516]
There appear to be cases where the formatter is not formatting items as strictly as it should be. There may also be inconsistencies with how files are formatted on different devices.

@myssto please provide any insights here if possible.

**Comment by hburn7** [comment]
Seems like we may be in luck if we at least flag these items as warnings. Also need to include more items in the .editorconfig itself. MSDN released a new default `.editorconfig` [here](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/code-style-rule-options)

**Comment by hburn7** [comment]
JetBrains IDEs (Rider) contain default code quality and formatting behavior. If our editorconfig doesn't specify something, the IDE rule is preferred. I guess the solution here is to expand the `.editorconfig` whenever this issue comes up, but users also need to be aware of what their IDE is doing formatting wise. As long as everything is formatted through `dotnet format` in the CLI, I don't think there's an issue?

**Comment by hburn7** [comment]
This is what I had to do to fix this issue.

![Image](https://github.com/user-attachments/assets/e7cc1d70-a296-40cc-bc0b-ad01bc03e19f)

**Comment by myssto** [comment]
> but users also need to be aware of what their IDE is doing formatting wise

not our problem. editorconfig should just be as verbose as possible

**Comment by hburn7** [comment]
I agree, there's just hundreds of jetbrains-specific rules that do get enforced at the IDE side. When a commit is made through the IDE I saw it runs the IDE formatting tool in addition to husky I think. It would probably make sense to put this in our docs somewhere.

**Comment by majorro** [comment]
I'd like to add that Rider goes weird for me with only this project... I haven't had any similar issues in other projects (with editorconfig or csharpier or anything else) Sometimes it forces something that isn't forced neither in editorconfig nor in internal settings.

For example this:
```
new()
{
    PropA = "asdfasdf",
    PropB = "fdasfsadfasdfa",
}
```
was forcibly reformatted to this:
```
new()
{
    PropA = "asdfasdf", PropB = "fdasfsadfasdfa",
}
```
even though object initializer line breaks are enabled both in editorconfig and Rider settings (with related ones here as well)
